include/roboteam_utils/HalfLine.h

```diff
@@ -6,31 +6,40 @@
 
 namespace rtt {
 /**
- * The HalfLine class represents a line that is semi-infinite long, i.e. it has a start point where the line starts and an direction in which it is infinite long.
+ * The HalfLine class represents a line that is semi-infinite long, i.e. it has a start point where the line starts and a direction in which it is infinite long.
  *
  * @author Haico Dorenbos
  * @since 2020-04-22
  */
 class HalfLine {
     private:
-        Vector2 start; // The location where the HalfLine starts
-        Vector2 direction; // This vector indicates in which direction the HalfLine is infinite long
+        Vector2 start; // The location where the HalfLine starts.
+        Vector2 goesThrough; // An arbitrary other vector (different than start) through which the HalfLine goes.
 
     public:
         /**
-         * Construct a new HalfLine object
-         * @param start The location where the HalfLine starts
-         * @param direction This vector indicates in which direction the HalfLine is infinite long
+         * Construct a new HalfLine object by using 2 different points on that HalfLine. Make sure that the given points are different.
+         * @param start The location where the HalfLine starts.
+         * @param goesThrough An arbitrary other vector through the HalfLine goes (make sure that it is different than the start point).
          */
-        constexpr HalfLine(const Vector2 &start, const Vector2 &direction) noexcept : start{start}, direction{direction} {};
+        explicit HalfLine(const Vector2 &start, const Vector2 &goesThrough);
```

Comment:
```suggestion
        contexpr HalfLine(const Vector2 &start, const Vector2 &goesThrough);
```

Doesn't have to be marked expliti if it has more than 2 arguments and constexpr ftw.

Reply:
Causes compile errors, so I won't change it...

Reply:
You should remove explicit though, unless you have a reason for it

Reply:
Fixed